crates/client-api/src/auth.rs

```diff
@@ -37,95 +39,91 @@ impl authorization::Credentials for SpacetimeCreds {
         if basic.username() != TOKEN_USERNAME {
             return None;
         }
-        Some(Self(basic))
+        let token = basic.password().to_owned();
+        Some(Self { token })
     }
     fn encode(&self) -> HeaderValue {
-        self.0.encode()
+        headers::Authorization::basic(TOKEN_USERNAME, &self.token).0.encode()
     }
 }
 
 impl SpacetimeCreds {
     pub fn token(&self) -> &str {
-        self.0.password()
+        &self.token
     }
     pub fn decode_token(&self, public_key: &DecodingKey) -> Result<SpacetimeIdentityClaims, JwtError> {
         decode_token(public_key, self.token()).map(|x| x.claims)
     }
     pub fn encode_token(private_key: &EncodingKey, identity: Identity) -> Result<Self, JwtError> {
         let token = encode_token(private_key, identity)?;
-        let headers::Authorization(basic) = headers::Authorization::basic(TOKEN_USERNAME, &token);
-        Ok(Self(basic))
+        Ok(Self { token })
+    }
+
+    fn from_request_parts(parts: &request::Parts) -> Result<Option<Self>, headers::Error> {
+        let res = match parts.headers.typed_try_get::<headers::Authorization<Self>>() {
+            Ok(Some(headers::Authorization(creds))) => return Ok(Some(creds)),
+            Ok(None) => Ok(()),
+            Err(e) => Err(e),
+        };
+        if let Ok(Query(creds)) = Query::<Self>::try_from_uri(&parts.uri) {
+            return Ok(Some(creds));
+        }
+        res.map(|()| None)
```

Comment:
Yeah, I guess, but then it would be a None that would never be Some. it doesn't much matter I suppose, it could be `Ok(None)`, the logic is just kinda convoluted and I was trying to preserve it while making it idiomatic.